src/test/osd/test_scrubber_be.cc

```diff
@@ -0,0 +1,647 @@
+    pending_inc.new_weight[i] = CEPH_OSD_IN;
+    pending_inc.new_uuid[i] = sample_uuid;
+  }
+  osdmap->apply_incremental(pending_inc);
+
+  // create a replicated pool
+  OSDMap::Incremental new_pool_inc(osdmap->get_epoch() + 1);
+  new_pool_inc.new_pool_max = osdmap->get_pool_max();
+  new_pool_inc.fsid = osdmap->get_fsid();
+  uint64_t pool_id = ++new_pool_inc.new_pool_max;
+  pg_pool_t empty;
+  auto p = new_pool_inc.get_new_pool(pool_id, &empty);
+  p->size = pconf.size;
+  p->set_pg_num(pconf.pg_num);
+  p->set_pgp_num(pconf.pgp_num);
+  p->type = pg_pool_t::TYPE_REPLICATED;
+  p->crush_rule = 0;
+  p->set_flag(pg_pool_t::FLAG_HASHPSPOOL);
+  new_pool_inc.new_pool_names[pool_id] = pconf.name;
+  osdmap->apply_incremental(new_pool_inc);
+  return osdmap;
+}
+
+pg_info_t TestTScrubberBe::setup_pg_in_map()
+{
+  pg_t rawpg(0, pool_id);
+  pg_t pgid = osdmap->raw_pg_to_pg(rawpg);
+  std::vector<int> up_osds;
+  int up_primary;
+
+  osdmap->pg_to_up_acting_osds(pgid,
+			       &up_osds,
+			       &up_primary,
+			       &acting_osds,
+			       &acting_primary);
+
+  std::cout << fmt::format(
+    "{}: pg: {} up_osds: {} up_primary: {} acting_osds: {} acting_primary: "
+    "{}\n",
+    __func__,
+    pgid,
+    up_osds,
+    up_primary,
+    acting_osds,
+    acting_primary);
+
+  spg = spg_t{pgid};
+  i_am = pg_shard_t{up_primary};
+  std::cout << fmt::format("{}: spg: {}  and I am {}\n", __func__, spg, i_am);
+
+  // the 'acting shards' set - the one actually used by the scrubber
+  std::for_each(acting_osds.begin(), acting_osds.end(), [&](int osd) {
+    acting_shards.insert(pg_shard_t{osd});
+  });
+  std::cout << fmt::format("{}: acting_shards: {}\n", __func__, acting_shards);
+
+  pg_info_t info;
+  info.pgid = spg;
+  /// \todo: handle the epochs:
+  // info.last_update = osdmap->get_epoch();
+  // info.last_complete = osdmap->get_epoch();
+  // info.last_osdmap_epoch = osdmap->get_epoch();
+  // info.history.last_epoch_marked_removed = osdmap->get_epoch();
+  info.last_user_version = 1;
+  info.purged_snaps = {};
+  info.last_user_version = 1;
+  info.history.last_epoch_clean = osdmap->get_epoch();
+  info.history.last_epoch_split = osdmap->get_epoch();
+  info.history.last_epoch_marked_full = osdmap->get_epoch();
+  info.last_backfill = hobject_t::get_max();
+  info.stats.stats.sum.num_objects_degraded = 0;
+  info.stats.stats.sum.num_objects_misplaced = 0;
+  info.stats.stats.sum.num_objects_unfound = 0;
+  info.stats.stats.sum.num_objects = 0;
+  info.stats.stats.sum.num_object_clones = 0;
+  info.stats.stats.sum.num_object_copies = 0;
+  info.stats.stats.sum.num_objects_missing_on_primary = 0;
+  info.stats.stats.sum.num_objects_degraded = 0;
+  info.stats.stats.sum.num_objects_misplaced = 0;
+  info.stats.stats.sum.num_objects_unfound = 0;
+  // info.stats.stats.sum.num_bytes_used = 0; // FIXME: find the right value
+  info.stats.stats.sum.num_bytes = 0;
+  info.stats.stats.sum.num_objects = 0;
```

Comment:
`object_stat_sum_t()` constructor sets zero to it's members.

Reply:
Thanks. (And later on I will modify osd_types.h to unify the initializations - currently some in the member dec and some in the ctor)